scripts/generators/es_template.py

```diff
@@ -204,23 +184,39 @@ def mapping_settings(mapping_settings_file):
     return mappings
 
 
-def template_settings(es_version, ecs_version, mappings_section, template_settings_file):
+def template_settings(ecs_version, template_version, mappings_section, template_settings_file, component_names=None):
```

Comment:
nit: instead of `template_version`, you could use a `bool` kwarg like `legacy=False`. Avoids having function calls with `1` and `2` magic numbers.

Simplifies the later v1/v2 check:

```python
if legacy:
    template = default_v1_template_settings(ecs_version)
else:
    template = default_v2_template_settings(ecs_version)
```

Reply:
Great point, updating